specfile/specfile.py

```diff
@@ -120,18 +189,28 @@ def _dump_debug_info(self, message) -> None:
         )
 
     @staticmethod
-    def _read_lines(path: Path) -> Tuple[List[str], bool]:
-        content = path.read_text(encoding="utf8", errors="surrogateescape")
-        return content.splitlines(), content[-1] == "\n"
+    def _read_lines(file: IOBase) -> Tuple[List[str], bool]:
+        file.seek(0)
+        raw_content = file.read()
+        if isinstance(raw_content, str):
+            content = raw_content
+        else:
+            content = raw_content.decode(**ENCODING_ARGS)
+        return content.splitlines(), content.endswith("\n")
 
     @property
-    def path(self) -> Path:
+    def path(self) -> Optional[Path]:
         """Path to the spec file."""
-        return self._path
+        try:
+            return Path(self._file.name)
+        except AttributeError:
+            return None
 
     @path.setter
     def path(self, value: Union[Path, str]) -> None:
-        self._path = Path(value)
+        self._file.close()
+        new_path = Path(value)
+        self._file = new_path.open("r+", **ENCODING_ARGS)
```

Comment:
There are actually a few possible scenarios and questions here:
- `Specfile` has been instantiated with a file or a string and the user is now setting `path`
  - do we want to allow this at all?
  - if yes, should we just silently open a new file? should we close the old one?
  - what if `self._file.name == value`?
- `Specfile` has been instantiated with a path
  - do we want to open a new file even if the path hasn't changed?

Reply:
Ok, if you want to keep the current approach, there are two things:
- nothing should happen if the path hasn't changed, i.e. if `value == self.path`
- if `self._file` is e.g. `StringIO`, its buffer is discarded when it is closed, so a user could lose modifications (though one could ask why would they be setting path in such case), so perhaps it's better not to explicitly close the old file (if there is no reference to it outside of `Specfile` instance it will be closed implicitly)